### PR TITLE
STATISTICS-76: Max Implementation

### PR DESCRIPTION
**Pull request by ani5rudh**

### CODE REVIEW

File: commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MaxTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            Arguments.of(new double[] {0.0d, Double.NaN, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY}, Double.NaN),
+            Arguments.of(new double[] {+0.0d, -0.0d, 1.0, 3.14}, 3.14),
+            Arguments.of(new double[] {-0.0, +0.0}, 0.0),
+            Arguments.of(new double[] {0.0, -0.0}, 0.0),
+            Arguments.of(new double[] {0.0, +0.0}, 0.0),
+            Arguments.of(new double[] {1.2, -34.56, 456.789, -5678.9012}, 456.789),
+            Arguments.of(new double[] {-23467824, 23648, 2368, 23749, -23424, -23492, -92397747}, 23749),
+            Arguments.of(new double[] {0.0d, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.MIN_VALUE}, Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {0.0d, +0.0d, -0.0d, Double.POSITIVE_INFINITY, -Double.MIN_VALUE}, Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {0.0d, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.MIN_VALUE}, Double.POSITIVE_INFINITY)
```
Comment (kinow) [inline commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MaxTest.java:104]: Quickly compared the code in this PR with the existing Min code, and everything looks OK. The only part I noticed that did not change was this with the `Double.MIN_VALUE`'s. Is that intentional or were we supposed to use `Double.MAX_VALUE` for this test?

**Review by kinow — COMMENTED** [review]
Change looks good to me already! Left one minor question. Thanks @ani5rudh !

File: commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MaxTest.java (hunk 1 above)
Comment (ani5rudh) [reply commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MaxTest.java]: Thanks @kinow, I've updated the tests for Max per your feedback.

**Review by kinow — APPROVED** [review]

File: commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MaxTest.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void testMax(double[] values, double expected) {
+        Max stat = Max.create();
+        Arrays.stream(values).forEach(stat);
+        double actual = stat.getAsDouble();
+        Assertions.assertEquals(expected, actual, "max");
+        Assertions.assertEquals(expected, Max.of(values).getAsDouble(), "max");
```
Comment (aherbert) [inline commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MaxTest.java:84]: The assertion message here is not helpful to distinguish this case. You could use "of(values)". Just change the test data so the test fails and you can see how JUnit will output the message. This should be helpful to diagnose the problem with the code.

**Review by aherbert — APPROVED** [review]
Looks good. Just a small note about useful assertion messages. Ideally these should allow you to know what part of the code went wrong.

You can squash all commits and update.

**Pull request merged**